test/controllers/sections/consignee/ConsigneeExportControllerSpec.scala

```diff
@@ -42,112 +44,175 @@ class ConsigneeExportControllerSpec extends SpecBase with MockUserAnswersService
-      val userAnswers = emptyUserAnswers.set(ConsigneeExportPage, true)
+        val userAnswers = emptyUserAnswers.set(ConsigneeExportPage, true)
 
-      val application = applicationBuilder(userAnswers = Some(userAnswers)).build()
+        val application = applicationBuilder(userAnswers = Some(userAnswers)).build()
 
-      running(application) {
-        val request = FakeRequest(GET, consigneeExportRoute)
+        running(application) {
+          val request = FakeRequest(GET, consigneeExportRoute)
 
-        val view = application.injector.instanceOf[ConsigneeExportView]
+          val view = application.injector.instanceOf[ConsigneeExportView]
 
-        val result = route(application, request).value
+          val result = route(application, request).value
 
-        status(result) mustEqual OK
-        contentAsString(result) mustEqual view(form.fill(true), NormalMode)(dataRequest(request), messages(application)).toString
+          status(result) mustEqual OK
+          contentAsString(result) mustEqual view(form.fill(true), NormalMode)(dataRequest(request), messages(application)).toString
+        }
       }
     }
 
-    "must redirect to the next page when valid data is submitted" in {
+    "onSubmit" - {
+
+      "must redirect to the next page when valid data is submitted - data is new" in {
+
+        val application =
+          applicationBuilder(userAnswers = Some(emptyUserAnswers))
+            .overrides(
+              bind[ConsigneeNavigator].toInstance(new FakeConsigneeNavigator(onwardRoute)),
+              bind[UserAnswersService].toInstance(mockUserAnswersService)
+            )
+            .build()
 
-      val userAnswers = emptyUserAnswers.set(ConsigneeExportPage, true)
+        running(application) {
 
-      val application =
-        applicationBuilder(userAnswers = Some(userAnswers))
-          .overrides(
-            bind[ConsigneeNavigator].toInstance(new FakeConsigneeNavigator(onwardRoute)),
-            bind[UserAnswersService].toInstance(mockUserAnswersService)
-          )
-          .build()
+          MockUserAnswersService.set().returns(Future.successful(emptyUserAnswers))
 
-      running(application) {
-        val request =
-          FakeRequest(POST, consigneeExportRoute)
-            .withFormUrlEncodedBody(("value", "true"))
+          val request =
+            FakeRequest(POST, consigneeExportRoute)
+              .withFormUrlEncodedBody(("value", "true"))
 
-        val result = route(application, request).value
+          val result = route(application, request).value
 
-        status(result) mustEqual SEE_OTHER
-        redirectLocation(result).value mustEqual onwardRoute.url
+          status(result) mustEqual SEE_OTHER
+          redirectLocation(result).value mustEqual onwardRoute.url
+        }
       }
-    }
 
-    "must return a Bad Request and errors when invalid data is submitted" in {
+      "must redirect to the next page when valid data is submitted - data has changed" in {
```

Comment:
This could probably do with the extra questions that get cleansed when the answer changes being added. Then ensuring that only the pages that weren't removed are stored in the user answers onSubmit

Reply:
I realised that I want to clear the whole section if the answer changes at all, regardless if they enter `true` or `false`, so I'm refactoring to ensure that works. I'll make sure the mock is taking in the correct value to verify the correct values are being removed though.